src/averageAges.js

```diff
@@ -37,7 +42,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people
+      .filter(person => person.mother !== null)
+      .map(person => person.mother)
+    : people.filter(person => person.sex === 'f');
+
+  const mothers = people.filter(person => women.includes(person.name));
+  const averageMothersAges = mothers.map(person => person.died - person.born);
+  const ages = women.map(person => person.died - person.born);
+  const averageWomenAges = withChildren
```

Comment:
why did you check this two times?
You can just find women with children or all women, depending on the condition & calculate average age
Rewrite this